### PR TITLE
Fix a consistency issue with command line parsing of file names (use FileInfo instead of string)

### PR DESCRIPTION
**Pull request by fadimounir**

### CODE REVIEW

File: src/coreclr/src/tools/crossgen2/crossgen2/CommandLineOptions.cs
Comment (trylek) [inline src/coreclr/src/tools/crossgen2/crossgen2/CommandLineOptions.cs]: I believe that, for the sake of consistency, you should also rename the corresponding field JitPath to have the type FileInfo (and perhaps fix a few places in the code to use the .FullPath property when accessing the actual path string).

**Review by trylek — APPROVED** [review]
LGTM modulo the consistency issue between the JitPath field type and the field type declared in the argument definition - not sure how this works, that's actually one of the most annoying aspects of the new command-line parser - it has tons of automated logic to support kind of just about everything using a set of very cryptic rules and reflection introspection so that "everything just works but you're completely busted if something happens to not work because the rules are so complicated there's basically no way to figure out what went wrong".

File: src/installer/pkg/projects/netcoreapp/pkg/Microsoft.NETCore.App.Crossgen2.pkgproj
Comment (dagood) [inline src/installer/pkg/projects/netcoreapp/pkg/Microsoft.NETCore.App.Crossgen2.pkgproj]: I assume this is to allow it to roll forward onto any netcoreapp build?

Should it maybe use the current build's product version? Or is it specifically desired to be able to run the crossgen2 pack on top of earlier builds of netcoreapp?

Either way, seems like it'd be useful to add this info to the comment above.

Comment (fadimounir) [reply src/installer/pkg/projects/netcoreapp/pkg/Microsoft.NETCore.App.Crossgen2.pkgproj]: > I assume this is to allow it to roll forward onto any netcoreapp build

Yes. Ideally, when we ship, this should just be 5.0

> Should it maybe use the current build's product version?

When testing using my SDK repo which is sync'd to the latest in master, its runtime version `5.0.0-alpha.1.20057.6` was smaller than the crossgen2 package version `5.0.100-alpha1-015772` (also, I believe this version might have a typo and should have been 5.0.100-alpha.1.01577.2). The runtime version used by the SDK will always lag the latest version of the crossgen2 package that can be downloaded from the nuget feed.

Is there a better way of solving this issue?

Comment (dagood) [reply src/installer/pkg/projects/netcoreapp/pkg/Microsoft.NETCore.App.Crossgen2.pkgproj]: I believe that when SDK daily builds are set up properly for the crossgen2 pack, it should have data in its MSBuild files that make it use the same version of the crossgen2 pack as the runtime bundled with that SDK. (This is assuming that crossgen2 pack acquisition is driven through MSBuild commands, which I think should be the goal.)

I'm not 100% sure that's the design we're aiming for though. If we want to be more loosely coupled (allow using a framework pack that's *newer* than the runtime being used) then hard-coding a low value would make sense. It seems like this could make for some confusing runtime errors if new private surface area is used somehow, but maybe that's ok.

@nguerrera, do you have thoughts on this pack?

Comment (nguerrera) [reply src/installer/pkg/projects/netcoreapp/pkg/Microsoft.NETCore.App.Crossgen2.pkgproj]: I'm not sure.

Comment (nguerrera) [reply src/installer/pkg/projects/netcoreapp/pkg/Microsoft.NETCore.App.Crossgen2.pkgproj]: It doesn't seem like we need to block on this since this patch is changing one hard-coded version for another. It might be worth chatting a bit in person, I'm not sure I understood the bigger, longer term picture yet.

**Review by dagood — APPROVED** [review]

File: src/installer/pkg/projects/netcoreapp/pkg/Microsoft.NETCore.App.Crossgen2.pkgproj
Comment (fadimounir) [reply src/installer/pkg/projects/netcoreapp/pkg/Microsoft.NETCore.App.Crossgen2.pkgproj]: I want to have a FD package for dotnet5.0 that has a runtimeconfig that says "I run on version 5.0 and anything above". Longer term for future releases, we'll change it to become a self-contained package. The reason I'd like to have a FD package in dotnet5.0 is to reduce download size for the crossgen2 package, but maybe it's not worth the complexity here, and we should just make it a self-contained package? We can chat more in person.

**Pull request merged**
